<launch>

  <arg name="gzname" default="gazebo"/>
  <arg name="gzworld" default="'$(find ihmc_gazebo)/worlds/atlas_polaris.world'"/>
  <arg name="hand_suffix" default=""/>
  <arg name="extra_gazebo_args" default="-q -e dart" />
  <arg name="model_args" default="" />

  <param name="/atlas/time_to_unpin" type="double" value="0.1"/>
  <param name="/atlas/startup_mode" type="string" value="car_front"/>

  <!-- start gazebo with the Atlas -->
  <include file="$(find ihmc_gazebo)/launch/atlas_no_controllers.launch">
    <arg name="gzname" value="$(arg gzname)"/>
    <arg name="gzworld" value="$(arg gzworld)"/>
    <arg name="extra_gazebo_args" value="$(arg extra_gazebo_args)"/>
  </include>

  <!-- Robot Description -->
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find ihmc_gazebo)/urdf/ihmc_atlas_v5.urdf.xacro'" />
  <param name="robot_initial_pose/x"     value="0.10" type="double"/>
  <param name="robot_initial_pose/y"     value="-0.45" type="double"/>
  <param name="robot_initial_pose/z"     value="1.08" type="double"/>
  <param name="robot_initial_pose/roll"  value="0.0" type="double"/>
  <param name="robot_initial_pose/pitch" value="-0.1" type="double"/>
  <param name="robot_initial_pose/yaw"   value="-0.0" type="double"/>  
  
  <param name="/robot_initial_configuration/back_bkx"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/back_bky"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/back_bkz"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/hokuyo_joint"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/l_arm_elx"	value="0.498"	type="double"/>
  <param name="/robot_initial_configuration/l_arm_ely"	value="2.0"		type="double"/>
  <param name="/robot_initial_configuration/l_arm_shx"	value="0.1"		type="double"/>
  <param name="/robot_initial_configuration/l_arm_shy"	value="-1.1"		type="double"/>
  <param name="/robot_initial_configuration/l_arm_wrx"	value="-0.004"	type="double"/>
  <param name="/robot_initial_configuration/l_arm_wry"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/l_arm_wry2"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/l_leg_akx"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/l_leg_aky"	value="-0.2"	type="double"/>
  <param name="/robot_initial_configuration/l_leg_hpx"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/l_leg_hpy"	value="-1.3"	type="double"/>
  <param name="/robot_initial_configuration/l_leg_hpz"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/l_leg_kny"	value="1.4"		type="double"/>
  <param name="/robot_initial_configuration/neck_ry"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/r_arm_elx"	value="-0.498"	type="double"/>
  <param name="/robot_initial_configuration/r_arm_ely"	value="2.0"		type="double"/>
  <param name="/robot_initial_configuration/r_arm_shx"	value="-0.1"	type="double"/>
  <param name="/robot_initial_configuration/r_arm_shy"	value="1.1"		type="double"/>
  <param name="/robot_initial_configuration/r_arm_wrx"	value="0.004"	type="double"/>
  <param name="/robot_initial_configuration/r_arm_wry"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/r_arm_wry2"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/r_leg_akx"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/r_leg_aky"	value="-0.2"	type="double"/>
  <param name="/robot_initial_configuration/r_leg_hpx"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/r_leg_hpy"	value="-1.3"	type="double"/>
  <param name="/robot_initial_configuration/r_leg_hpz"	value="0.0"		type="double"/>
  <param name="/robot_initial_configuration/r_leg_kny"	value="1.4"		type="double"/> 
    
  <include file="$(find ihmc_gazebo)/launch/ihmc_atlas_v5_bringup.launch"/>
</launch>
